text/0000-monorepo.md

```diff
@@ -0,0 +1,359 @@
+
+Push access to the project would be given to all members of the GitHub teams included in the project.
+
+Publish access would ideally be automated through CI, but manual access can be granted to packages that belong to a specific GitHub team / workstream.
+
+## Add-ons
+
+Within this proposal, the role of an add-on will most likely need to be re-defined. Currently, an add-on exists as a supplement to the Carbon Design System. This means that teams arrive at solutions that they need that don't exist in Carbon currently, and they have a desire to share within their team. The add-on model works well for this direct use-case, but unfortunately when scaling out Carbon to all of IBM this model has a habit of entrenching behavior.
+
+For example, imagine we had teams A, B, and C all working on platforms. Team A comes up with a `PlatformHeader` component that teams B and C would like to use.
+
+Team B can use the component as-is, which is great. However, Team C would like some current functionality tweaked to meet their use-case. While teams A and C could work together to arrive at a joint resolution, often times deadlines and scope limit this kind of collaboration.
+
+In addition, the experience of the developer can be confusing in these moments. The packages for these add-ons would like be:
+
+```
+# Team A
+carbon-addons-a
+carbon-addons-b
+carbon-addons-c
+```
+
+In the existing landscape, team B would need to install (at minimum) the following:
+
+```bash
+yarn add carbon-components \
+  carbon-components-react \ 
+  carbon-icons \
+  carbon-addons-a \
+  carbon-addons-a-react \
+  carbon-addons-b \
+  carbon-addons-b-react 
+```
+
+Ideally, the best experience for this group would be a single `carbon-addons-b` or `carbon-addons-b-react`.
+
+The other downside to this is the aspect of who owns support and who decides what is in-scope for enhancements of a component. When components are built inside add-ons, this can fall on the team whose add-on houses the component. However, if another team is requesting enhancements most likely they should be the ones to implement (when reasonable).
+
+As a result, the proposal is that **all** component development happens in the Carbon repo and add-on packages move to becoming **the defacto way to use Carbon for a Business unit, portfolio, offering, or team**. What this means is that add-ons move increasingly towards their domain, and components like `PlatformHeader` should move to the main Carbon repo.
+
+This does not prohibit multiple `PlatformHeader` components, however. If we used the situation above, we could have a singular `platform-header` package. Inside this package we could have multiple implementations under `PlatformHeader.A.js`, `PlatformHeader.B.js`, etc. This allows us to both allow teams to ship things under their unique deadlines/constraints while also allowing colocation of related components with the eventual goal of converging implementations.
```

Comment:
Now does this also apply to core components? For example, `Search.js`, `Search.wsc.js`, `Search.cloud.js`?

This also introduces some interesting issues for angular ... I'm guessing we would end up with `search.module.ts`, `search-wsc.module.ts`, `search-cloud.module.ts`, with the core `search.module` being the only one exported from the index (but the others would consume it and export it from their addons).